### PR TITLE
transactions: initial transactions definition

### PR DESCRIPTION
**Pull request by cedric-parisi**
Simple transaction definition using gunk

The transaction model is quite empty for now as this PR is more to get started on transaction definition.

### CODE REVIEW

File: gunks/v1/transactions/transactions.gunk [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+type Transaction struct {
+	// TransactionID is the unique identifier of a transaction
+	TransactionID string `pb:"1" json:"transaction_id"`
+	// SourceAccountID is the identifier of the account emitting the transaction
+	SourceAccountID string `pb:"2" json:"source_account_id"`
+	// DestinationAccountID is the identifier of the account receiving the transaction
+	DestinationAccountID string `pb:"3" json:"destination_account_id"`
+}
+
+// GetTransactionRequest is the request enveloppe to get an transaction by its identifier
```
Comment (jwangsadinata) [inline gunks/v1/transactions/transactions.gunk]: s/enveloppe/envelope

File: gunks/v1/transactions/transactions.gunk [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+type GetTransactionRequest struct {
+	// TransactionID is the unique identifier of a transaction
+	TransactionID string `pb:"1" json:"transaction_id"`
+}
+
+// GetTransactionsRequest is the request enveloppe to get a list of transactions
```
Comment (jwangsadinata) [inline gunks/v1/transactions/transactions.gunk]: s/enveloppe/envelope

File: gunks/v1/transactions/transactions.gunk [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+// CreateTransactionRequest wraps all required field for a transaction creation
+type CreateTransactionRequest struct {
+	// SourceAccountID is the identifier of the account emitting the transaction
+	SourceAccountID string `pb:"1" json:"source_account_id"`
+	// DestinationAccountID is the identifier of the account receiving the transaction
+	DestinationAccountID string `pb:"2" json:"destination_account_id"`
+}
+
+// CreateTransactionResponse is the response enveloppe for a transaction creation
```
Comment (jwangsadinata) [inline gunks/v1/transactions/transactions.gunk]: s/enveloppe/envelope

File: gunks/v1/transactions/transactions.gunk [hunk 4]
```diff
@@ -0,0 +1,96 @@
+package transactions
+
+import (
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+	// "google/api/annotations.proto"
```
Comment (jwangsadinata) [inline gunks/v1/transactions/transactions.gunk]: why is this commented btw @cedric-parisi?

File: gunks/v1/transactions/transactions.gunk (hunk 1 above)
Comment (cedric-parisi) [reply gunks/v1/transactions/transactions.gunk]: Yes I realised in accounts that I've used french spelling all the way

File: gunks/v1/transactions/transactions.gunk (hunk 4 above)
Comment (cedric-parisi) [reply gunks/v1/transactions/transactions.gunk]: these lines were added by gunk convert. I removed them as they are not used by gunk generate (which is the command we'll use)

File: gunk/v1/transactions/transactions.gunk [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	//
+	// +gunk http.Match{
+	//		Method:	"POST",
+	// 		Path:	"/v1/transactions",
+	// 		Body:	"*",
+	//	}
+	// +gunk openapiv2.Operation{
+	//		Tags: []string{"Transaction"},
+	//		Description: "Creates a new transaction and returns its id",
+	//		Summary: "Create a transaction",
```
Comment (mvdan) [inline gunk/v1/transactions/transactions.gunk]: format?

**Comment by cedric-parisi** [comment]
I'm closing this stale PR and open a new one. Sorry for the inconvenience 

https://github.com/openbank/gunk/pull/7

**Pull request closed without merging**